Add firewall controllers

Bigzhuang opened the pull request:
including below CRDs:
1. rule
2. zone
3. snat
4. dnat
5. forwarding

Commit:
Add firewall controllers
including below CRDs:
1. rule
2. zone
3. snat
4. dnat
5. forwarding

Bigzhuang committed:
Merge branch 'master' into Add-firewall-CRd

cheng1li commented on platform/crd-ctrlr/src/api/v1alpha1/firewalldnat_types.go:
```diff
@@ -0,0 +1,74 @@
+}
+
+// FirewallDNATStatus defines the observed state of FirewallDNAT
+type FirewallDNATStatus struct {
```
Seems the `FirewallDNATStatus ` struct is not used, so remove it

cheng1li commented on platform/crd-ctrlr/src/api/v1alpha1/firewallforwarding_types.go:
```diff
@@ -0,0 +1,64 @@
+}
+
+// FirewallForwardingStatus defines the observed state of FirewallForwarding
+type FirewallForwardingStatus struct {
```
Also, remove this

xuyizhox committed:
Remove Status defination in all *types.go

add check logic if error.code equal to 404 in cnf.DeleteObject(handler, instance) of  base_controller.go
for case that cr failed to be deleted due to finalizer exists.

xuyizhox committed:
add indexer for pods owned by cnf to allow cnf deployment yaml without label sdewanPurpose in tempalte filed
